### ISSUE TITLE
Enable backtraces on receipt of SIGSEGV

### ISSUE THREAD

**Issue by krader1961** [att/ast#144]
We've recently identified numerous instances of using variables before they are initialized. This often leads to a SIGSEGV being sent to the shell. Most users don't have core dumps enabled and it's not easy to arrange for core dumps in Travis CI. So write a basic stack backtrace when SIGSEGV is received. This is what allowed me to find the root cause of issue #142. And as I type this the Travis OpenSUSE environment is failing thusly:

```
test builtins(C.UTF-8) begins at 2017-11-25+17:49:43
TODO: Enable these tests when the custom builtins of external commands are available.
TODO: Skipping test - 'cd with no arguments fails if HOME is unset'. It should be fixed later.
shtests[333]: eval: line 1: 11399: Memory fault(coredump)
test builtins(C.UTF-8) failed at 2017-11-25+17:49:56 with exit code 267 [ 195 tests 267 errors ]
```

**Comment by krader1961** [comment]
With this enhancement this is what the segfault reveals on Travis OpenSUSE:

```
test builtins(C.UTF-8) begins at 2017-11-25+22:01:50
TODO: Enable these tests when the custom builtins of external commands are available.
TODO: Skipping test - 'cd with no arguments fails if HOME is unset'. It should be fixed later.
0   dump_backtrace + 50
1   handle_sigsegv + 26
2   /lib64/libc.so.6(+0x34950) [0x7f87d6216950]
3   vmregion + 169
4   _ast_free + 198
5   /source/arch/linux.i386-64/bin/ksh() [0x57f06b]
6   sfsetbuf + 4302
7   /source/arch/linux.i386-64/bin/ksh() [0x55f996]
8   /lib64/libc.so.6(+0x37139) [0x7f87d6219139]
9   /lib64/libc.so.6(+0x37185) [0x7f87d6219185]
10  /source/arch/linux.i386-64/bin/ksh() [0x4202d7]
11  /source/arch/linux.i386-64/bin/ksh() [0x419499]
12  sh_main + 3222
13  main + 37
14  __libc_start_main + 245
15  _start + 41
shtests[333]: eval: line 1: 11383: Memory fault(coredump)
test builtins(C.UTF-8) failed at 2017-11-25+22:02:04 with exit code 267 [ 195 tests 267 errors ]
```

Just a guess but this looks like a double-free bug. Or perhaps freeing a block that wasn't malloc'd --- possibly due to an uninitialized pointer var containing a random value.

**Comment by siteshwar** [comment]
Thanks for the pull request. Just as a sidenote, if you want to run any tests under ```valgrind``` you can use ```--wrapper``` parameter for meson. For e.g.

```
meson test --wrapper valgrind builtins  
```

**Comment by krader1961** [comment]
Implemented by commit f03b9bac.
